packages/lib/modules/pool/pool.tokens.display.ts

```diff
@@ -0,0 +1,108 @@
+import { GqlNestedPool, GqlPoolBase } from '@repo/lib/shared/services/api/generated/graphql'
+import { Pool } from './PoolProvider'
+import { ApiToken, ApiTokenWithBalance, PoolCore, PoolListItem } from './pool.types'
+// import { isV3Pool } from './pool.helpers'
+import { PoolToken } from '../tokens/token.helpers'
+import { isSameAddress } from '@repo/lib/shared/utils/addresses'
+import { Address } from 'viem'
+import { sortBy } from 'lodash'
+
+// CRITICAL TODO: duplicated function as importing from pool.helpers creates a circular dependency or something
+// (use the import from pool.helpers to see the error)
+export function isV3Pool(
+  pool: Pick<Pool | PoolListItem | GqlPoolBase, 'protocolVersion'>
+): boolean {
+  return pool.protocolVersion === 3
+}
+
+export function getCompositionDisplayTokens(pool: PoolCore | GqlNestedPool): ApiTokenWithBalance[] {
+  const tokens = getPoolTokens(pool).map(token => {
+    if (token.hasNestedPool && token.nestedPool) {
+      return {
+        ...token,
+        nestedTokens: getCompositionDisplayTokens(token.nestedPool as GqlNestedPool),
+      } as ApiTokenWithBalance
+    }
+    return token as ApiTokenWithBalance
+  })
+
+  return sortApiTokensBySymbol(
+    excludeNestedBptTokens(tokens as ApiToken[], pool.address)
+  ) as ApiTokenWithBalance[]
+}
+
+export function getHeaderDisplayTokens(pool: PoolCore): ApiToken[] {
```

Comment:
I'm not sure Header is the right word because it's for the pool header and list. Trying to describe them here: They are reference tokens for the user so that they know at a glance what can be added to the pool. maybe?

```suggestion
// The set of tokens that are reference tokens for the user so that they know at a glance what can be added to the pool. Used in the pool header and in the pools list.
export function getUserReferenceTokens(pool: PoolCore): ApiToken[] {
```